libs/platform/ebpf_platform.h

```diff
@@ -379,15 +387,15 @@ extern "C"
      * @returns The previous lock_state required for unlock.
      */
     _Requires_lock_not_held_(*lock) _Acquires_lock_(*lock) _IRQL_requires_max_(DISPATCH_LEVEL) _IRQL_saves_
-        _IRQL_raises_(DISPATCH_LEVEL) ebpf_lock_state_t ebpf_lock_lock(_Inout_ ebpf_lock_t* lock);
+        _IRQL_raises_(DISPATCH_LEVEL) ebpf_lock_state_t ebpf_lock_lock(_Inout_ const ebpf_lock_t* lock);
 
     /**
      * @brief Release exclusive access to the lock.
      * @param[in, out] lock Pointer to memory location that contains the lock.
      * @param[in] state The state returned from ebpf_lock_lock.
      */
     _Requires_lock_held_(*lock) _Releases_lock_(*lock) _IRQL_requires_(DISPATCH_LEVEL) void ebpf_lock_unlock(
-        _Inout_ ebpf_lock_t* lock, _IRQL_restores_ ebpf_lock_state_t state);
+        _Inout_ const ebpf_lock_t* lock, _IRQL_restores_ ebpf_lock_state_t state);
```

Comment:
How can it be Inout if it's const?

Reply:
Not quite sure how to handle this case. There are paths where we need to read state from an object protected by a lock., but acquiring the lock is essentially a mutable operation. Unfortunately, C doesn't appear to have a way to mark the lock field as mutable, so either we cast at the ebpf_lock_lock call-site or we cast it internally.